### ISSUE TITLE
checkers: make typeSwitchVar generate 1 warn per switch

### ISSUE THREAD

**Issue by quasilyte** [go-critic/go-critic#778]
Right now, it prints a warning for every case inside a type switch that does type assertion.
Should warn about type switch only once and, probably, tell how many cases can benefit from that re-write.

```diff
- case 0 can benefit from type switch var
- case 1 can benefit from type switch var
- case 3 can benefit from type switch var
+ 3 cases can benefit from type switch var
```
Checker sources: `/checkers/typeSwitchVar_checker.go`.
Relevant tests are in `/checkers/testdata/typeSwitchVar/`.

Run tests: 
```bash
go test -v -run=/typeSwitchVar github.com/go-critic/go-critic/checkers
```
After unit tests are fixed, integration tests may require adjastments as well.
The entire `go test` (without `-run`) must pass.
